osxcollector/output_filters/summary_filters/html.py

```diff
@@ -327,4 +329,11 @@ def _print_val(self, val):
                 self._write('</li>')
             self._write('</ul>')
         elif isinstance(val, basestring) or isinstance(val, Number):
+            self._encode_val(val)
+
+    def _encode_val(self, val):
+        try:
             self._write(u'<span class="val">{0}</span>'.format(val))
+        except UnicodeDecodeError as err:
+            sys.stderr.write('Unicode decode error while processing the value: {0}\n{1}\n'.format(val, err))
+            self._write('<span class="val">{0}</span>'.format(val))
```

Comment:
I see now. I would still say replace is better to call ``unicode( .. errors='replace')``, otherwise you will end up with mixed encoding, potential control characters, and doubly logged errors. Anything that fails UnicodeDecodeError here will also hit the exception in _write. And the output will then contain utf-8 AND raw bytes, which browsers can probably handle most of the time, but I would try to keep it all unicode.